mesheryctl/internal/cli/root/system/model.go

```diff
@@ -0,0 +1,364 @@
+	"os"
+	"strings"
+
+	"github.com/layer5io/meshery/mesheryctl/internal/cli/root/config"
+	"github.com/layer5io/meshery/mesheryctl/pkg/utils"
+	"github.com/layer5io/meshery/server/models"
+	"github.com/layer5io/meshkit/models/meshmodel/core/v1alpha1"
+	"github.com/manifoldco/promptui"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"gopkg.in/yaml.v2"
+)
+
+var (
+	// flag used to specify format of output of view {model-name} command
+	outFormatFlag string
+)
+
+// represents the `mesheryctl system model list` subcommand.
+var listModelCmd = &cobra.Command{
+	Use:   "list",
+	Short: "list models",
+	Long:  "list name of all registered models",
+	Example: `
+// View current provider
+mesheryctl system model list
+	`,
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		//Check prerequisite
+		hcOptions := &HealthCheckOptions{
+			IsPreRunE:  true,
+			PrintLogs:  false,
+			Subcommand: cmd.Use,
+		}
+		hc, err := NewHealthChecker(hcOptions)
+		if err != nil {
+			return ErrHealthCheckFailed(err)
+		}
+		// execute healthchecks
+		err = hc.RunPreflightHealthChecks()
+		if err != nil {
+			cmd.SilenceUsage = true
+			return err
+		}
+		cfg, err := config.GetMesheryCtl(viper.GetViper())
+		if err != nil {
+			return err
+		}
+		ctx, err := cfg.GetCurrentContext()
+		if err != nil {
+			return err
+		}
+		err = ctx.ValidateVersion()
+		if err != nil {
+			return err
+		}
+		return nil
+	},
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if len(args) != 0 {
+			return errors.New(utils.SystemModelSubError("this command takes no arguments\n", "list"))
+		}
+		mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
+		if err != nil {
+			log.Fatalln(err, "error processing config")
+		}
+
+		baseUrl := mctlCfg.GetBaseMesheryURL()
+
+		url := fmt.Sprintf("%s/api/meshmodels/models?pagesize=all", baseUrl)
+		req, err := utils.NewRequest(http.MethodGet, url, nil)
+		if err != nil {
+			utils.Log.Error(ErrGettingRequestContext(err))
+			return err
+		}
+
+		client := &http.Client{}
+		resp, err := client.Do(req)
+		if err != nil {
+			utils.Log.Error(ErrConnectingToServer(err))
+			return err
+		}
+
+		// defers the closing of the response body after its use, ensuring that the resources are properly released.
+		defer resp.Body.Close()
+
+		data, err := io.ReadAll(resp.Body)
+		if err != nil {
+			utils.Log.Error(ErrInvalidAPIResponse(err))
+			return err
+		}
+
+		modelsResponse := &models.MeshmodelsAPIResponse{}
+		err = json.Unmarshal(data, modelsResponse)
+		if err != nil {
+			utils.Log.Error(ErrUnmarshallingAPIData(err))
+			return err
+		}
+
+		header := []string{"Model Name"}
+		rows := [][]string{}
+
+		for _, model := range modelsResponse.Models {
+			if len(model.DisplayName) > 0 {
+				// in `mesheryctl system model view [model-name]` command `model.Name` must be
+				// passed in order to fetch its details
+				rows = append(rows, []string{model.Name})
+			}
```

Comment:
```suggestion
		header := []string{"Category Name", "Model Name", "Version"}
		rows := [][]string{}

		for _, model := range modelsResponse.Models {
			if len(model.DisplayName) > 0 {
				// in `mesheryctl system model view [model-name]` command `model.Name` must be
				// passed in order to fetch its details
				rows = append(rows, []string{model.Category.Name, model.Name, model.Version})
			}
		}
```
Further more details needed for the user as in the some category there are more than one models with different version

Reply:
at the moment we've flow as when user runs `mesheryctl system model view [model-name]` command, if there are more than one models for the given name it shows selector UI (as it is in `mesheryctl login` for provider) for all available versions, if we add version in `model list` command's output then we will have to add a flag in `model view` command for the specification fo version, so in which way do I implement it?.

Reply:
@suhail34 what I think though about it is that giving selector UI is more convenient and intuitive for user.

Reply:
This is not for view command it is for list command

Reply:
@leecalcote any suggestion here `mesheryctl system model list` will be listing just model name that are some time repeating because of different version don't you think it should be showing category , model name and version that would give more insights